client/src/App.jsx

```diff
@@ -118,7 +126,7 @@ class App extends Component {
         })
         .catch((err) => {
           if (err.response.status === 500) {
-            window.location.href = '/error';
+            window.location.href = <serverError />;
```

Comment:
Why did you do that!?
It is not possible to assign a React component to a `window.location.href`
What we're doing here `window.location.href = '/error';` is converting the current window location to an endpoint called `/error` which is handled by React Router Dom for displaying a 500 - Server Error component